Add docs for resizing clickhouse and postgres (#2515)
PostHog/posthog.com: contents/docs/self-host/runbook/clickhouse/resize-disk.md

---
title: Resize disk
sidebar: Docs
showTitle: true
---

#### How to increase disk size

1. Connect to the Clickhouse container to verify the data directory filesystem size (in this example 20G)
    ```shell
    kubectl -n posthog exec -it chi-posthog-posthog-0-0-0 -- /bin/bash
    clickhouse@chi-posthog-posthog-0-0-0:/$ df -h /var/lib/clickhouse/
    Filesystem                                                                Size  Used Avail Use% Mounted on

>     /dev/disk/by-id/scsi-0DO_Volume_pvc-f39035c1-c68c-4572-81f2-273de6eb088c   20G   50M   19G   1% /var/lib/clickhouse
> 

Author: ```suggestion
    /dev/disk/by-id/scsi-0DO_Volume_pvc-f39035c1-c68c-4572-81f2-273de6eb088c   20G   50M   19G   1% /var/lib/clickhouse
    ```

```

---
title: Resize disk
sidebar: Docs
showTitle: true
---

> #### How to increase disk size

Reviewer: Is this runbook slightly different than the one for Kafka/PostgreSQL? The wording seems different and I don't see the prerequisite section

Author: It's different as you don't need to do anything other than change the values file

Reviewer: Ok but prerequisite should still be here

Author: I assumed it's always expandable because of this https://github.com/PostHog/charts-clickhouse/blob/main/charts/posthog/templates/clickhouse_instance.yaml#L15 no?

Reviewer: That's technically not enough (it maybe works for AWS and GCP but what about the other clouds?)

1. Connect to the Clickhouse container to verify the data directory filesystem size (in this example 20G)
    ```shell
    kubectl -n posthog exec -it chi-posthog-posthog-0-0-0 -- /bin/bash
    clickhouse@chi-posthog-posthog-0-0-0:/$ df -h /var/lib/clickhouse/
    Filesystem                                                                Size  Used Avail Use% Mounted on
    /dev/disk/by-id/scsi-0DO_Volume_pvc-f39035c1-c68c-4572-81f2-273de6eb088c   20G   50M   19G   1% /var/lib/clickhouse
    ```

1. In your Helm chart configuration, update the `clickhouseOperator.storage` value in `value.yaml` to the target size (40G in this example)

1. Run a `helm` upgrade

> 1. Connect to the Kafka container to verify the new filesystem size

Reviewer: ```suggestion
1. Connect to the ClickHouse container to verify the new filesystem size
```

    ```shell
    kubectl -n posthog exec -it chi-posthog-posthog-0-0-0 -- /bin/bash
    clickhouse@chi-posthog-posthog-0-0-0:/$ df -h /var/lib/clickhouse/
    Filesystem                                                                Size  Used Avail Use% Mounted on
    /dev/disk/by-id/scsi-0DO_Volume_pvc-f39035c1-c68c-4572-81f2-273de6eb088c   40G  186M   38G   1% /var/lib/clickhouse
    ```

> Note that this doesn't work for decreases, if you try the disk won't be resized and the following errors can be seen in the clickhouse operator pod

Reviewer: ```suggestion
Note: this procedure doesn't work to a volume. If you try, the disk won't be resized and the following errors can be seen in the ClickHouse operator pod
```

Author: ```suggestion
Note: this procedure doesn't work to decrease a volume. If you try, the disk won't be resized and the following errors can be seen in the ClickHouse operator pod
```
